content/guide/core/file-system.md

```diff
@@ -0,0 +1,747 @@
+
+### Deleting a folder
+
+To delete a folder, use the `remove` or `removeSync` method:
+
+```ts
+folder.remove().then((value: boolean)=>{
+
+console.log(value)
+
+}).catch((error)=>{
+
+console.error(error.message)
+})
+```
+
+### Creating, writing to and reading from a text file
+<!-- Preview: https://stackblitz.com/edit/nativescript-stackblitz-templates-5yndru?file=app/main-view-model.ts -->
+- To create a file, call the `getFile()` method on an instance of the `Folder` class and pass it the file name with the extension.
+
+```ts
+folder.getFile("my-text.txt")
+//or 
+const filePath = path.join(folder.path, 'my-text.txt')
+const file = File.fromPath(filePath)
+```
+
+- To save a text to a file, use the `writeText` or `writeTextSync` method:
+
+```ts
+file
+  .writeText('Some text')
+  .then(result => {
+    // Succeeded writing to the file.
+ 
+  })
+  .catch(error => {
+    console.log(error)
+  })
+
+```
+- To read data from a text file, use the `readText` or `readTextSync` method of the file instance:
+```ts
+   file.readText().then(res => {
+      // Succeeded read from file.
+      
+    })
+    .catch(error => {
+
+  })
+```
+### Check if a file exists
+
+To check if a file exists, call the static [exists](#exists) method of the File class and pass it the file path string:
+
+```ts
+const exists = File.exists(fPath)
+```
+
+### Renaming a file
+
+To rename a file, use the `rename` or `renameSync` method: 
+
+```ts
+file.rename("new-name.ext").then((value)=>{
+
+}).catch(err=>{
+
+})
+```
+
+### Saving a binary data to a file
+<!-- TODO: binary data example -->
+To save binary data to a file, use the `write` or `writeSync` method:
+
+```ts
+file
+  .write(binary data)
+  .then(result => {
+    // Succeeded writing to the file.
+ 
+  })
+  .catch(error => {
+    console.log(error)
+  })
+
+```
+
+
+### Reading a binary data from a file
+
+To read binary data, use the `read` or `readSync` method of the File instance. For example, you can save an image on the device and retrive it as follows:
+
+```ts
+async readAnImage() {
+
+    const folder = knownFolders.documents()
+    const fPath = path.join(folder.path, 'cat.png')
+    const imageFile = File.fromPath(fPath)
+
+    try {
+
+      const imageSource = await ImageSource.fromFile('~/assets/images/download.jpeg')
+      const saved: boolean = imageSource.saveToFile(fPath, "png")
+
+      if (saved) {
+        // READ BINARY DATA
+        const imageBinary = await imageFile.read()
+        const img = await ImageSource.fromData(imageBinary)
+      }
+
+    } catch (err) {
+
+    }
+}
+```
+
+### Deleting a file
+
+To remove a file, use the [remove](#remove) or [removeSync](#removesync) method of the File instance:
+
+```ts
+file
+  .remove()
+  .then((res: boolean) => {
+    // Success removing the file.
+  })
+  .catch(err => {
+  })
+ ``` 
+
+ ### Normalizing a path
+
+ To normalize a path use the `normalize` or create it with the `join` method from the [path](#path-operations) object:
```

Comment:
To normalize a path use the `normalize` method or create the path with the `join` method from the [path](#path-operations) object: